zetta_utils/db_annotations/collection.py

```diff
@@ -0,0 +1,55 @@
+"""Manage annotation collections in a DB Layer."""
+
+from __future__ import annotations
+
+import time
+import uuid
+
+from zetta_utils.layer.db_layer import DBRowDataT, build_db_layer
+from zetta_utils.layer.db_layer.datastore import DatastoreBackend
+
+from . import constants
+
+DB_NAME = "collections"
+INDEXED_COLS = ("name", "created_by", "created_at", "modified_by", "modified_at")
+NON_INDEXED_COLS = ("comment",)
+
+DB_BACKEND = DatastoreBackend(DB_NAME, project=constants.PROJECT, database=constants.DATABASE)
+DB_BACKEND.exclude_from_indexes = NON_INDEXED_COLS
+COLLECTIONS_DB = build_db_layer(DB_BACKEND)
+
+
+def read_collection(collection_id: str) -> DBRowDataT:
```

Comment:
Might be possible to have a more narrow type here, but for mypy to not complain we'll have to assert that the indexed item is indeed conforming to the more narrow type, or just use `cast`